src/components/SearchPanelControls.js

```diff
@@ -13,11 +13,23 @@ export class SearchPanelControls extends Component {
   constructor(props) {
     super(props);
 
-    this.state = { search: '' };
+    this.state = { search: props.query };
     this.handleChange = this.handleChange.bind(this);
     this.submitSearch = this.submitSearch.bind(this);
   }
 
+  /**
+   * Set the component's local search state
+   * to blank when the query has been cleared
+   */
+  componentDidUpdate(prevProps) {
+    const { query } = this.props;
+
+    if (query === '' && query !== prevProps.query) {
+      this.handleChange({ target: { value: '' } });
```

Comment:
`eslint` complained about it due to [this rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-did-update-set-state.md) (which maybe means all I'm doing is fooling the linter)...

I would be happy to talk through how we could achieve the same thing (or disable the rule for this line 🤷‍♂)

Reply:
Yeah.. it feels like its starting to get sketchy.. since we are also controlling the state internally. Two options I can think of ( I think I prefer the first)

 - Just disable the rule explicitly inline (maybe helps us know we are doing something sketchy)
 
 or

 - Attach the query / search state up to redux for everything.. (seems sketchy since we will be dispatching on every change)

Reply:
Yeah, I went down the hooking up query state to redux for everything, but that ended up causing complications where we would need to create the search in state (which is what `REQUEST_SEARCH` does) before we've actually searched (which seems like it could have other elbow effects).

I'll disable and add a comment about what's going on.